### PR TITLE
Nomination Pools migration v5: RewardPool fix

### PR DESCRIPTION
**Pull request by rossbulat**
Companion for https://github.com/paritytech/substrate/pull/13715

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/2591057

File: runtime/polkadot/src/lib.rs [hunk 1]
```diff
@@ -1418,6 +1418,7 @@ pub type Migrations = (
 		Runtime,
 		NominationPoolsMigrationV4OldPallet,
 	>,
+	pallet_nomination_pools::migration::v5::MigrateToV5<Runtime>,
```
Comment (kianenigma) [inline runtime/polkadot/src/lib.rs:1421]: won't `MigrateToV4` require an `allow(deprecated)`?

**Review by kianenigma — APPROVED** [review]

File: runtime/polkadot/src/lib.rs (hunk 1 above)
Comment (ggwpez) [reply runtime/polkadot/src/lib.rs:1421]: Yes, will replace with `MigrateV3ToV5` which does both.

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/2593601

**Review by bkchr — APPROVED** [review]

**Comment by ggwpez** [comment]
bot merge

**Comment by rossbulat** [comment]
bot rebase

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/2595679

**Comment by rossbulat** [comment]
bot merge

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/2595960

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/2596479

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/2596657

**Comment by ggwpez** [comment]
bot merge

**Pull request merged**